src/main/java/org/jenkinsci/plugins/ghprb/GhprbTrigger.java

```diff
@@ -371,14 +364,12 @@ public String getGitHubAuthId() {
     }
     
     public GhprbGitHubAuth getGitHubApiAuth() {
-        if (gitHubAuthId == null) {
-            for (GhprbGitHubAuth auth: getDescriptor().getGithubAuth()){
-                gitHubAuthId = auth.getId();
-                getDescriptor().save();
-                return auth;
-            }
+        GhprbGitHubAuth auth = getDescriptor().getGitHubAuth(gitHubAuthId);
+        if (gitHubAuthId == null && auth != null) {
+            gitHubAuthId = auth.getId();
+            getDescriptor().save();
         }
-        return getDescriptor().getGitHubAuth(gitHubAuthId);
+        return auth;
```

Comment:
The changes to both this 'getGitHubApiAuth()` and `getGitHubAuth(String gitHubAuthId)` now prevent the possibility of an exception being thrown if the `List<GhprbGitHubAuth>` is empty.  (It would have thrown an exception before).